player/configfiles.c

```diff
@@ -215,8 +215,12 @@ static char *mp_get_playback_resume_config_filename(struct MPContext *mpctx,
     char *res = NULL;
     void *tmp = talloc_new(NULL);
     const char *path = NULL;
-    if (mp_is_url(bstr0(fname))) {
-        path = fname;
+    bstr play_path = bstr0(fname);
+    // mpv:// should be transparent to the user
+    if (bstr_eatstart0(&play_path, "mpv://"))
+        fname = play_path.start;
```

Comment:
Not sure I get it. Why does the prefix need to be specially handled from a watch later file?

Reply:
watch later file is using the "input" name directly, for example youtube link without processing. And the same for mpv:// happens, but we want it to be transparent between mpv:// and not, so we strip it here.

Reply:
> I also don't get it

Watch later is special egg, it doesn't use `filename`, `stream_open_filename` or any other value that is actually used for playback. It uses the one that user provided in playlist, hance why it already has other normalization there. This is I presume done to make watch later work as close to user provided filenames, not something processed by the hooks.

> Shouldn't it be handled at stream layer instead, like stream_mpv.c that does all these things internally?

What things? How?

Reply:
Shouldn't the `mpv://` be recognized and handled like any other existing protocol? I assume this already works for any other stream type.

Reply:
> Look into `demux_playlist`. `mpv://` is basially a special case of a playlist of 1 item. Several streams like `edl` are just dummy wrappers which forwards handling in demuxers.

I think you are mixing demuxers with streams.

Reply:
@na-na-hi: I've made a draft. Is this what you requested? It's still doesn't work, because I need to implement track selection.

For me it is completely not pragmatic, to have whole mpv demuxer, that spawns timelines with single demuxer, just because we don't want to eatstart0. Also I'm not sure it will work for every source, especially once without duration.

Reply:
playlist makes sense, if we have to go through demuxer.